api/views.py

```diff
@@ -230,6 +231,36 @@ class DeviceApiPicture(generics.RetrieveDestroyAPIView):
     queryset = Picture.objects.all()
     serializer_class = PictureSerializer
 
+class DeviceApiPictureRotate(generics.RetrieveUpdateAPIView):
+    model = Picture
+    queryset = Picture.objects.all()
+    serializer_class = PictureSerializer
+
+    def patch(self, request, *args, **kwargs):
+        from PIL import Image
+        import StringIO
+        from django.core.files.uploadedfile import InMemoryUploadedFile
+
+
+        picture = get_object_or_404(Picture, pk=self.kwargs["pk"])
+        img = Image.open(picture.image)
+
+        #determine if orientation is left or right
+        if self.kwargs["orientation"] == "right":
+            img  = img.transpose(Image.ROTATE_270)
+        if self.kwargs["orientation"] == "left":
+            img  = img.transpose(Image.ROTATE_90)
+        img_io = StringIO.StringIO()
+        img.save(img_io, format = 'JPEG')
+        img_file = InMemoryUploadedFile(img_io, None, 'foo.jpg', 'image/jpeg',
+                                  img_io.len, None)
+        #delete old image
+        picture.image.delete(save = True)
+        picture.image = img_file
+        picture.save()
+        return Response(status=status.HTTP_200_OK)
```

Comment:
`status=200` should also work, but details..

Reply:
So should I change it or not?

Reply:
http://python.net/~goodger/projects/pycon/2007/idiomatic/handout.html#the-zen-of-python-1

Reply:
But they also say "Try to make your programs easy to read and obvious. " :P But I'll change it ^^